segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -167,6 +172,56 @@ public void setupAppend(SetupAppend setupAppend) {
                 });
     }
 
+    @VisibleForTesting
+    CompletableFuture<Void> setupTokenExpiryTask(SetupAppend setupAppend) {
+        String segment = setupAppend.getSegment();
+        UUID writerId = setupAppend.getWriterId();
+        long requestId = setupAppend.getRequestId();
+
+        Duration expiryDuration = this.durationToExpiry(setupAppend);
+
+        if (expiryDuration == null) {
+            return CompletableFuture.completedFuture(null);
+        } else {
+            return Futures.delayedTask(() -> {
+                if (isSetupAppendCompleted(segment, writerId)) {
+                    try {
+                        log.trace("Informing writer {} that sent request {}, about token expiry for segment {}",
+                                writerId, requestId, segment);
+                        if (!connection.isClosed()) {
+                            connection.close();
+                        }
+                    } catch (RuntimeException e) {
+                        // Log and ignore
+                        log.warn("Unable to inform writer {} that sent request {}, about token expiry for segment {}",
```

Comment:
We can improve/remove the warn message here since we are closing the connection and not informing the writer via a Wirecommand.

Reply:
We don't need to catch the exception anymore after we modified the code here to close the connection, so I've removed the code.